core/deployment/src/main/java/io/quarkus/runner/RuntimeClassLoader.java

```diff
@@ -76,7 +82,29 @@
     public RuntimeClassLoader(ClassLoader parent, List<Path> applicationClassesDirectories, Path frameworkClassesDirectory,
             Path transformerCache) {
         super(parent);
-        this.applicationClassDirectories = applicationClassesDirectories;
+        List<Path> nonEmptyAppClasses = new ArrayList<>();
+        try {
+            Map<String, Path> applicationClasses = new HashMap<>();
+            for (Path i : applicationClassesDirectories) {
+                if (Files.isDirectory(i)) {
+                    if (Files.list(i).count() > 0) {
+                        nonEmptyAppClasses.add(i);
```

Comment:
Wondering if we should consider only the ones with classes in them? Not sure it's worth it though.

Reply:
These can also be used for resources, but this has much less of an impact on hot reload time